### ISSUE TITLE
3.0.41 introduces false positive of MA0042

### ISSUE THREAD

**Issue by Timovzl** [meziantou/Meziantou.Analyzer#1101]
### Version of the Meziantou.Analyzer NuGet package

3.0.41

### Rule Identifier

MA0042, MA0190

### Target Framework

net10.0

### C# Language version

14

### Description

I've noticed a bug that appears in all current versions starting with 3.0.41. MA0042 (use async overload) has started reporting false positives in a specific scenario.

I'm using TUnit and doing some assertions like this:

```cs
await Assert.That(result.Id).IsNotDefault();
```

The analyzers tells me to "use `That instead of `That`. Applying the fix produces this:

```cs
await (await Assert.That(result.Id)).IsNotDefault();
```

The latter is rightfully reject with: "'ValueAssertion<string>' does not contain a definition for 'GetAwaiter'".

Interestingly, the problem appears for a nongeneric overload: `ValueAssertion<string> That(string? value, [CallerArgumentExpression(nameof(value))] string? expression = null)`. The generic overloads do not suffer from it.

### Reproduction Steps

Use TUnit and write a test method:

```c#
[Test]
public async Task MutateTo_WithLoanForNonReturnable_ShouldThrow()
{
	var x = "Blah";
	await Assert.That(x).IsEqualTo("Blah");
}
```
